src/compiler/checker.ts

```diff
@@ -9618,7 +9624,7 @@ namespace ts {
         }
 
         function getLiteralTypeFromProperties(type: Type, include: TypeFlags) {
-            return getUnionType(map(getPropertiesOfType(type), t => getLiteralTypeFromProperty(t, include)));
+            return getUnionType(map(getPropertiesOfType(type), p => getLiteralTypeFromProperty(p, include)));

```

Comment:
Rename seems unneeded :P

Reply:
`p` is a property, not a type. Seems weird to name it `t`.